engine/src/quicktime.cpp

```diff
@@ -551,19 +553,19 @@ void MCQTGetRecordLoudness(integer_t &r_loudness)
         
 intenum_t MCQTGetRecordFormatId(MCStringRef p_string)
 {
-    for (int i = 0; i < sizeof(record_formats) / sizeof(record_formats[0]); i++)
+    for (auto t_iter = std::begin(record_formats); t_iter != std::end(record_formats); ++t_iter)
     {
-        if (MCStringIsEqualToCString(p_string, record_formats[i].label, kMCCompareCaseless))
-            return record_formats[i].value;
+        if (MCStringIsEqualToCString(p_string, t_iter.label, kMCCompareCaseless))
+            return t_iter.value;
```

Comment:
There are two exciting C++11 features that you could use here!

You could use a range-based `for` loop:

```c++
for (auto&& t_format : record_formats) {
    if (MCStringIsEqualToCString(p_string, t_format.label, kMCCompareCaseless))
        return t_iter.value;
}
return 0;
```

Or you could use `std::find_if`:

```c++
auto t_found = std::find_if(std::begin(record_formats), std::end(record_formats),
    [&](const record_format& p_fmt) {
        return MCStringIsEqualToCString(p_string, p_fmt.label, kMCCompareCaseless);
    });
return (t_found != std::end(record_formats) ? t_found->value : 0);
```

Arguably range-based `for` gives the nicer code of the two, but YMMV. 👍 

Note that some of these lookup functions don't seem to handle the "not found" case by returning a default value.